### DISCUSSION TITLE
How to load multiple nested tables without using LoadWith primary table again?

### DISCUSSION THREAD

**Discussion by sgashua** [linq2db/linq2db#4242] [Q&A] [answered]
My Table B links to many tables.

I currently use like this
```
context.TableA
       .LoadWith(x => x.TableB)
            .ThenLoad(x => x.TableC)
                 .ThenLoad(x => x.TableD)
       .LoadWith(x => x.TableB)
            .ThenLoad(x => x.TableE)
       .LoadWith(x => x.TableB)
            .ThenLoad(x => x.TableF)
```

Any shortcut linq to get all nested tables just with 1 line of LoadWith(x => x.TableB)?

**Comment by sdanyliv** [comment, marked as answer]
It depends on which properties are collections. Simple variant is `.LoadWith(x => x.TableB.TableC.TableD)`. If `TableB` is collection `.LoadWith(x => x.TableB.First().TableC.TableD)`
